test/e2e-shared-tests/helpers.go

```diff
@@ -0,0 +1,266 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func assertKernelArgsEqual(t *testing.T, cs *framework.ClientSet, target corev1.Node, startKargs string, expectedKargs []string) {
+	t.Helper()
+
+	kargs := getKernelArgs(t, cs, target)
+
+	for _, karg := range expectedKargs {
+		assert.Contains(t, kargs, karg, "Missing karg %s", karg)
+	}
+
+	assert.Equal(t, kargs, startKargs)
+}
+
+func getKernelArgs(t *testing.T, cs *framework.ClientSet, target corev1.Node) string {
+	return helpers.ExecCmdOnNode(t, cs, target, "chroot", "/rootfs", "/usr/bin/rpm-ostree", "kargs")
+}
+
+func waitForConfigDriftMonitorStart(t *testing.T, cs *framework.ClientSet, node corev1.Node) {
+	t.Helper()
+
+	// With the way that the Config Drift Monitor is wired into the MCD,
+	// "machineconfiguration.openshift.io/state" gets set to "Done" before the
+	// Config Drift Monitor is started. This is fine, however it makes testing a
+	// bit tricky as we rely upon "machineconfiguration.openshift.io/state" to be
+	// "Done" before we commence our test.
+	//
+	// Because of this, we must infer that the Config Drift Monitor is started by
+	// looking at the MCD pod logs, hence this function. It's also worth noting
+	// that the Config Drift Monitor text may not immediately appear in the log,
+	// so we have to poll for it.
+
+	start := time.Now()
+
+	err := wait.PollImmediate(2*time.Second, 1*time.Minute, func() (bool, error) {
+		mcdPod, err := helpers.MCDForNode(cs, &node)
+		require.Nil(t, err)
+
+		logs, err := cs.Pods(mcdPod.Namespace).GetLogs(mcdPod.Name, &corev1.PodLogOptions{
+			Container: "machine-config-daemon",
+		}).DoRaw(context.TODO())
+		require.Nil(t, err)
+
+		return strings.Contains(string(logs), configDriftMonitorStartupMsg), nil
+	})
+
+	end := time.Since(start)
+
+	if err == nil {
+		t.Logf("Config Drift Monitor is running (waited %v)", end)
+	}
+
+	require.Nil(t, err, "expected config drift monitor to start (waited %v)", end)
+}
+
+func assertNodeAndMCPIsDegraded(t *testing.T, cs *framework.ClientSet, node corev1.Node, mcp mcfgv1.MachineConfigPool, filename string) {
+	t.Helper()
+
+	logEntry := fmt.Sprintf("content mismatch for file \"%s\"", filename)
+
+	// Assert that the node eventually reaches a Degraded state and has the
+	// config mismatch as the reason
+	t.Log("Verifying node becomes degraded")
+
+	assertNodeReachesState(t, cs, node, func(n corev1.Node) bool {
+		isDegraded := n.Annotations[constants.MachineConfigDaemonStateAnnotationKey] == string(constants.MachineConfigDaemonStateDegraded)
+		hasReason := strings.Contains(n.Annotations[constants.MachineConfigDaemonReasonAnnotationKey], logEntry)
+		return isDegraded && hasReason
+	})
+
+	mcdPod, err := helpers.MCDForNode(cs, &node)
+	require.Nil(t, err)
+
+	assertLogsContain(t, cs, mcdPod, logEntry)
+
+	// Assert that the MachineConfigPool eventually reaches a degraded state and has the config mismatch as the reason.
+	t.Log("Verifying MachineConfigPool becomes degraded")
```

Comment:
Request: when looking thru the logs we don't explicitly say config mismatch detected or anything referencing config,  [example logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/2795/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/1466917418067562496/artifacts/e2e-gcp-op/test/build-log.txt). Can we add something around here which references config drift or config mismatch?

```suggestion
	t.Log("Verifying MachineConfigPool becomes degraded due to config mismatch")
```

Reply:
For sure. Great idea!